src/rust/session/src/lib.rs

```diff
@@ -51,6 +61,19 @@ pub struct Session {
     write_buffer: Buffer,
     min_capacity: usize,
     max_capacity: usize,
+    // hold current interest set
+    interest: Interest,
+    // TODO(bmartin): consider moving these fields and associated logic
+    // out into a response tracking struct. It would make the session
+    // type more applicable to clients if we move this out.
+    //
+    /// a timestamp which is used to calculate response latency
+    timestamp: Instant,
+    /// this is a queue of pending response sizes
+    pending_responses: VecDeque<usize>,
+    /// this holds the total response bytes pending so we can know how many
```

Comment:
If I understand correctly, this records the bytes pending _before_ the last call to `compose`, hence we can compare the current write buffer bytes and this value to get the  size of the last response. The comment can be made more clear on how this value is being used.